src/ext/readerFileDesc.c

```diff
@@ -74,19 +65,28 @@ RdbxReaderFileDesc *RDBX_createReaderFileDesc(RdbParser *p, int fd, int fdCloseW
     if (flags==-1) {
 
         RDB_reportError(p, RDB_ERR_FAILED_GET_FD_FLAGS,
-            "RDBX_createReaderFileDesc(): Error getting file descriptor flags");
+            "RDBX_createReaderFileDesc(fd=%d): Error getting file descriptor flags", fd);
         return NULL;
     }
 
     if (flags & O_NONBLOCK) {
         RDB_reportError(p, RDB_ERR_NONBLOCKING_FD,
-            "RDBX_createReaderFileDesc(): fd must be set to blocking mode");
+            "RDBX_createReaderFileDesc(fd=%d): fd must be set to blocking mode", fd);
+        return NULL;
+    }
+
+    FILE *file = fdopen(fd, "r");
+
+    if (file == NULL) {
+        RDB_reportError(p, RDB_ERR_NONBLOCKING_FD,
```

Comment:
Is this error code accurate? This error is not actually about blocking/nonblocking fd. It looked a bit confusing to me. 
Btw, I'd also print `strerror(errno)` here to figure out why `fdopen()` failed.